File: chapter-11/ios-specific/NativeModuleApp/ios/NativeModuleApp/HelloManager.m
//
//  HelloManager.m
//  NativeModuleApp
//
//  Created by Daniel Ward on 6/16/18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "HelloManager.h"

@implementation HelloManager
RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(
  greetUser: (NSString *)name isAdmin:(BOOL *)isAdmin callback: (RCTResponseSenderBlock) callback
) {
  NSString *greeting =
    [NSString stringWithFormat:
     @"Welcome %@, you %@ an administrator.", name, isAdmin ? @"are" : @"are not"];

  callback(@[greeting]);
}

@end

File: chapter-11/android-specific/NativeModuleApp/android/app/src/main/java/com/nativemoduleapp/HelloManager.java
package com.nativemoduleapp;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class HelloManager extends ReactContextBaseJavaModule {
  public HelloManager(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  public String getName() {
    return "HelloManager";
  }

  @ReactMethod
  public void greetUser(String name, Boolean isAdmin, Callback callback) {
    System.out.println("User Name: " + name + ", Administrator: " + (isAdmin ? "Yes" : "No"));
    String greeting = "Welcome " + name + ", you " + (isAdmin ? "are" : "are not") + " an administrator";

    callback.invoke(greeting);
  }
}
